gnomad_constraint/plots/generate_constraint_plots.R

```diff
@@ -243,3 +257,161 @@ for (version in names(datasets)) {
+####################################################################
+# Filter to MANE Select Ensembl transcripts that are present in both v2 and v4
+# and that have no constraint flags
+comparison_df <- filter_transcripts(constraint_data, "v4") %>% filter(.data$v2 == TRUE)
+
+comparison_df <- comparison_df %>%
+  select(
+    all_of(
+      c(
+        "transcript",
+        "v2",
+        "v4",
+        "lof.oe_ci.upper.v2",
+        "lof.oe_ci.upper.v4",
+        "lof.z_score.v2",
+        "lof.z_score.v4",
+        "lof.oe.v2",
+        "lof.oe.v4"
+      )
+    )
+  )
+
+# Reformat the data for plotting
+comparison_df <- reformat_for_metric_comparison(comparison_df)
+
+comparison_plot <- plot_metric_comparison(comparison_df, use_presentation_sizes=use_presentation_sizes)
+comparison_plot
+comparison_plot
+plot_path <- get_plot_path(
+  "v2_v4_compare_values",
+  output_basedir = output_basedir
+)
+
+ggsave(
+  comparison_plot,
+  filename = plot_path,
+  dpi = 300,
+  width = 8,
+  height = 4,
+  units = "in"
+)
+
+
+####################################################################
+####################################################################
+# Plot observed vs expected values
+####################################################################
+####################################################################
+for (version in versions_to_plot) {
+  # Filter to preferred Ensembl transcripts (mane select or canonical) without any constraint flags
+  filtered_data <- filter_transcripts(
+    constraint_data,
+    version,
+    preferred_transcripts_only = TRUE,
+    enst_only = TRUE,
+    include_flagged_transcripts = FALSE
+  )
+  
+  # Reformat the data for plotting
+  oe_data <- reformat_for_observed_vs_expected(filtered_data, version)
+  # Plot observed vs expected values
+  oe_plot <- plot_observed_vs_expected(oe_data, version)
+
+  plot_path <- get_plot_path(
+    "obs_vs_exp",
+    version = version,
+    output_basedir = output_basedir
+  )
+  ggsave(oe_plot, filename = plot_path, dpi = 300, width = 7, height = 6, units = "in")
+}
+
+
+####################################################################
+####################################################################
+# Plot proportion observed vs mu for v4
+####################################################################
+####################################################################
+training_data <- load_constraint_metrics(
+  version = "v4",
+  output_basedir = output_basedir,
+  training_set = TRUE,
+  release = FALSE,
+  public = FALSE,
+)
+
+
+# Remove version suffix from the training data to prepare for plotting
+training_data <- remove_version_suffix(training_data)
+
+
+data_with_predictions <- get_predicted_proportion_observed(df = training_data,
+                                                           coverage_metric=coverage_metric,
+                                                           high_coverage_cutoff=high_coverage_cutoff)
```

Comment:
where is `high_coverage_cutoff` defined?

Reply:
accidentally deleted, will add back in